assets/js/components/teams/TeamShow.jsx

```diff
@@ -130,38 +113,49 @@ class TeamShow extends Component {
           membership={this.state.editingMembership}
           updateMembership={updateMembership}
         />
-      </DashboardLayout>
+      </div>
     )
-  }
-}
 
-function mapStateToProps(state, ownProps) {
-  const currentTeamId = state.auth.currentTeamId
-  const team = state.entities.teams[currentTeamId]
+    const activityView = (
+      <Paper className={classes.paper}>
+        <header className={classes.header}>
+          <Typography variant="headline" component="h3">
+            Team History
+          </Typography>
+        </header>
 
-  let memberships = []
-  if (team !== undefined) {
-    memberships = Object
-      .values(state.entities.memberships)
-  }
+        <AuditTable />
+      </Paper>
+    )
 
-  let invitations = []
-  if (team !== undefined) {
-    invitations = Object
-      .values(state.entities.invitations)
-      .filter(invitation => invitation.pending)
-  }
+    const tabs = [{
+      label: "Access View",
+      content: accessView,
+      path: "/teams/access",
+    }, {
+      label: "Activity View",
+      content: activityView,
+      path: "/teams/activity",
+    }]
 
-  return {
-    currentTeamId,
-    memberships,
-    invitations
+    return (
+      <div>
+        <UserCan type="not" action="view" itemType="auditTrails">
```

Comment:
Added UserCan with type of "not". Let me know if you like a separate component instead.

Also, with this setup, there's a visual bug when you refresh the browser on /teams/activity. It flashes the members and invitations tables briefly due user being null in the beginning. Ive tried many things and think we must have the user injected into the component through mapStateToProps, which allows us to look for when the user.role == null to prevent the flashing of the member and invite tables.

Reply:
we have access to the user in the `<UserCan>` component right? can we just prevent rendering if role is null?